iceoryx2-cxx/include/iox2/entry_handle_mut.hpp

```diff
@@ -0,0 +1,136 @@
+
+    /// Updates the value by copying the passed value into it.
+    void update_with_copy(ValueType value);
+
+    /// Consumes the [`EntryHandleMut`] and loans an uninitialized entry value that can be used to update without copy.
+    template <ServiceType ST, typename KeyT, typename ValueT>
+    friend auto loan_uninit(EntryHandleMut<ST, KeyT, ValueT>&& self) -> EntryValueUninit<ST, KeyT, ValueT>;
+
+    /// Returns an ID corresponding to the entry which can be used in an event based communication
+    /// setup.
+    auto entry_id() const -> EventId;
+
+  private:
+    template <ServiceType, typename>
+    friend class Writer;
+    template <ServiceType ST, typename KeyT, typename ValueT>
+    friend auto update(EntryValue<ST, KeyT, ValueT>&& self) -> EntryHandleMut<ST, KeyT, ValueT>;
+    template <ServiceType ST, typename KeyT, typename ValueT>
+    friend auto discard(EntryValue<ST, KeyT, ValueT>&& self) -> EntryHandleMut<ST, KeyT, ValueT>;
+    template <ServiceType ST, typename KeyT, typename ValueT>
+    friend auto discard(EntryValueUninit<ST, KeyT, ValueT>&& self) -> EntryHandleMut<ST, KeyT, ValueT>;
+
+    explicit EntryHandleMut(iox2_entry_handle_mut_h handle);
+
+    void drop();
+
+    auto take_handle_ownership() -> iox2_entry_handle_mut_h;
+
+    iox2_entry_handle_mut_h m_handle = nullptr;
+};
+
+template <ServiceType S, typename KeyType, typename ValueType>
+inline EntryHandleMut<S, KeyType, ValueType>::EntryHandleMut(iox2_entry_handle_mut_h handle)
+    : m_handle { handle } {
+}
+
+template <ServiceType S, typename KeyType, typename ValueType>
+inline EntryHandleMut<S, KeyType, ValueType>::EntryHandleMut(EntryHandleMut&& rhs) noexcept {
+    *this = std::move(rhs);
+}
+
+template <ServiceType S, typename KeyType, typename ValueType>
+inline auto EntryHandleMut<S, KeyType, ValueType>::operator=(EntryHandleMut&& rhs) noexcept -> EntryHandleMut& {
+    if (this != &rhs) {
+        drop();
+        m_handle = std::move(rhs.m_handle);
+        rhs.m_handle = nullptr;
+    }
+
+    return *this;
+}
+
+template <ServiceType S, typename KeyType, typename ValueType>
+inline EntryHandleMut<S, KeyType, ValueType>::~EntryHandleMut() noexcept {
+    drop();
+}
+
+template <ServiceType S, typename KeyType, typename ValueType>
+inline void EntryHandleMut<S, KeyType, ValueType>::update_with_copy(ValueType value) {
+    // NOLINTNEXTLINE(cppcoreguidelines-owning-memory): required by C API
+    auto value_ptr = new ValueType(value);
+    iox2_entry_handle_mut_update_with_copy(&m_handle, value_ptr, sizeof(ValueType), alignof(ValueType));
+    // NOLINTNEXTLINE(cppcoreguidelines-owning-memory): required by C API
+    delete value_ptr;
+}
+
+template <ServiceType S, typename KeyType, typename ValueType>
+inline auto loan_uninit(EntryHandleMut<S, KeyType, ValueType>&& self) -> EntryValueUninit<S, KeyType, ValueType> {
+    EntryValueUninit<S, KeyType, ValueType> entry_value_uninit;
```

Comment:
Only the `EntryValueUninit` should know how it can be initialized. Could you please adjust the private constructor so that it has one additional argument, the `iox2_entry_handle_mut` and then this ctor calls ` iox2_entry_handle_mut_loan_uninit`

When during construction something fails we have to search for the actual construction since it is not happening in the ctor.
Also if we need to construct this object in some place else, we have to repeat the construction logic and then this can easily diverge.